### PR TITLE
Fix loading of 3D chart settings

### PR DESCRIPTION
**Pull request by antoniusriha**
There were still some bugs with loading the settings of 3D charts. Details are in the commit messages.

### CODE REVIEW

File: src/Charts/ModelWindow.cpp [hunk 1]
```diff
@@ -321,18 +322,19 @@ static t_preset *presets;
...
-        zSelector->setCurrentIndex(presets[index].z);
-        colorSelector->setCurrentIndex(presets[index].color);
-        ignore->setChecked(presets[index].ignore);
-        binWidthSlider->setValue(presets[index].bin);
-        binWidthLineEdit->setText(QString("%1").arg(presets[index].bin));
-
-        setDirty();
-        if (index) setData(true);
-    }
+    if (index < 0 || !context->tab->chartsettings()->isVisible())
```
Comment (liversedge) [inline src/Charts/ModelWindow.cpp]: this is not a good approach; you are intertwining gui state with configuration; they shouldn't be related this way.

Comment (antoniusriha) [inline src/Charts/ModelWindow.cpp]: Guess you're right. But I don't know how else to fix it. What I'd like to do, is: I only want applyPreset to run, if the user selected a preset and not if the preset was set by the QT property loading engine. The isVisible flag seemed to tell best, whether it was the user or something else.

**Comment by antoniusriha** [comment]
I've now employed another approach: The `currentIndexChanged` signal is only connected to `applyPreset` after the `preset` property has been loaded. I guess that is not the cleanest solution either, but better as the first one?

**Comment by liversedge** [comment]
thats probably the best way of fixing it

**Pull request closed without merging**